### PR TITLE
Refactor: commands

### PR DESCRIPTION
**Pull request by DimaDemchenko**

### CODE REVIEW

File: packages/p2p-media-loader-core/src/core.ts [hunk 1]
```diff
@@ -36,15 +36,15 @@ export class Core<TStream extends Stream = Stream> {
...
     simultaneousP2PDownloads: 3,
     highDemandTimeWindow: 15,
     httpDownloadTimeWindow: 3000,
     p2pDownloadTimeWindow: 6000,
     webRtcMaxMessageSize: 64 * 1024 - 1,
-    p2pNotReceivingBytesTimeoutMs: 1000,
+    p2pNotReceivingBytesTimeoutMs: 2000,
     p2pInactiveLoaderDestroyTimeoutMs: 30 * 1000,
-    httpNotReceivingBytesTimeoutMs: 1000,
+    httpNotReceivingBytesTimeoutMs: 3000,
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/core.ts:47]: Please retest the new default values

File: packages/p2p-media-loader-core/src/p2p/commands/types.ts [hunk 2]
```diff
@@ -17,11 +17,13 @@ export type PeerSegmentCommand = BasePeerCommand<
   | PeerCommandType.SegmentDataSendingCompleted
 > & {
   i: number; // segment id
+  r?: number; // request id
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/p2p/commands/types.ts:20]: Make `r` required for all the commands

File: packages/p2p-media-loader-core/src/p2p/peer.ts [hunk 3]
```diff
@@ -166,7 +179,10 @@ export class Peer {
       }
 
       case PeerCommandType.SegmentAbsent:
-        if (this.downloadingContext?.request.segment.externalId === command.i) {
+        if (
+          this.downloadingContext?.request.segment.externalId === command.i ||
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/p2p/peer.ts:183]: Change to `&&`

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by mrlika — APPROVED** [review]

**Comment by Chocobozzz** [comment]
Hi,

Can you detail this protocol change? Is the purpose to resend a P2P request to the same peer for the same segment?

**Comment by mrlika** [comment]
Hi, @Chocobozzz,

There were issues handling segment requests via P2P when a segmented download was canceled and restarted immediately. To address this, a randomly generated request ID is now added to the segment request command.

Fortunately, the library is designed so that P2P protocol version changes are minor because the protocol version is added to the WebTorrent info hash: https://github.com/Novage/p2p-media-loader/blob/0972c005958df549f3b27b8da0b6bf721450969c/packages/p2p-media-loader-core/src/utils/stream.ts#L23

This means that swarms consist of peers of the same protocol version. If peers of different protocol versions download the same stream, they will be in separate swarms, eliminating the ability for non-compatible communications.

**Comment by Chocobozzz** [comment]
Thank you for your answer @mrlika :pray: 

Yes I was aware about the protocol version, we correctly handle it :) https://github.com/Chocobozzz/PeerTube/blob/fbee171a0bea725e74ddc9ca4a00875c2250afd4/server/core/lib/hls.ts#L27

**Pull request merged**